codebundles/k8s-kubectl-namespace-healthcheck/sli.robot

```diff
@@ -0,0 +1,114 @@
+    ...    pattern=\w*
+    ...    example=Normal|Warning
+    ...    default=Warning
+    ${EVENT_AGE}=    RW.Core.Import User Variable    EVENT_AGE
+    ...    type=string
+    ...    description=The time window in minutes as to when the event was last seen.
+    ...    pattern=\w*
+    ...    example=5m
+    ...    default=5m
+    ${EVENT_THRESHOLD}=    RW.Core.Import User Variable    EVENT_THRESHOLD
+    ...    type=string
+    ...    description=The maximum total events to be still considered healthy. 
+    ...    pattern=\w*
+    ...    example=2
+    ...    default=0
+    ${CONTAINER_RESTART_AGE}=    RW.Core.Import User Variable    CONTAINER_RESTART_AGE
+    ...    type=string
+    ...    description=The time window in minutes as search for container restarts.
+    ...    pattern=\w*
+    ...    example=5m
+    ...    default=5m
+    ${CONTAINER_RESTART_THRESHOLD}=    RW.Core.Import User Variable    CONTAINER_RESTART_THRESHOLD
+    ...    type=string
+    ...    description=The maximum total container restarts to be still considered healthy. 
+    ...    pattern=\w*
+    ...    example=2
+    ...    default=0
+    ${binary_name}=    RW.K8s.Get Binary Name    ${DISTRIBUTION}
+    Set Suite Variable    ${binary_name}    ${binary_name}
+
+*** Tasks ***
+
+Get Event Count and Score
+    ${event_count}=    RW.K8s.Count Events By Age and Type
+    ...    namespace=${NAMESPACE}
+    ...    context=${CONTEXT}
+    ...    kubeconfig=${kubeconfig}
+    ...    target_service=${kubectl}
+    ...    binary_name=${binary_name}
+    ...    event_age=${EVENT_AGE}
+    ...    event_type=${EVENT_TYPE}
+    Log    ${event_count} total events found with event type ${event_type} up to age ${event_age}
+    ${event_score}=    Evaluate    1 if ${event_count} <= ${EVENT_THRESHOLD} else 0
+    Set Global Variable    ${event_score}
+
+Get Container Restarts and Score
+    ${container_restart_count}=    RW.K8s.Count Container Restarts By Age
+    ...    namespace=${NAMESPACE}
+    ...    context=${CONTEXT}
+    ...    kubeconfig=${kubeconfig}
+    ...    target_service=${kubectl}
+    ...    binary_name=${binary_name}
+    ...    container_restart_age=${CONTAINER_RESTART_AGE}
+    Log    ${container_restart_count} total container restarts found in the last ${CONTAINER_RESTART_AGE}
+    ${container_restart_score}=    Evaluate    1 if ${container_restart_count} <= ${CONTAINER_RESTART_THRESHOLD} else 0
+    Set Global Variable    ${container_restart_score}
+
+Get NotReady Pods
+    ${pods_notready_count}=    RW.K8s.Count Notready Pods
+    ...    namespace=${NAMESPACE}
+    ...    context=${CONTEXT}
+    ...    kubeconfig=${kubeconfig}
+    ...    target_service=${kubectl}
+    ...    binary_name=${binary_name}
+    Log    ${pods_notready_count} total unready pods
+    ${pods_notready_score}=    Evaluate    1 if ${pods_notready_count} == 0 else 0
+    Set Global Variable    ${pods_notready_score}
+
+Generate Namspace Score
```

Comment:
Ahh, warning here - robot does not garuntee the ordering of task stanzas by default. We only enforce it in tasksets if the user sets `DependsOn` and that's explicitly set when invoking the file.

So there's a chance `Generate Namespace Score` could run first

Reply:
To clarify - you probably need to mash all these together into 1 task

Reply:
In reading this: http://robotframework.org/robotframework/latest/RobotFrameworkUserGuide.html#execution-order (and other docs) it implies that they are always executed as defined in the file. My experience in testing the codebundle is consistent with this assumption. 

I don't particularly mind squishing it all into a single task, but had considered that using multiple tasks allows us to (at a later date) roll up the per-task statistics, which may be helpful in troubleshooting or observing particularly long running tasks (esp in the case of processing large json payload outputs). 

Have you run into issues where somehow the tasks weren't executed in order?

Reply:
Additionally, I just ran a test where one task fails, and the final score fails too. I think this is a decent behaviour since it won't push a result that is missing data.